### PR TITLE
[AzureMonitorExporter] logging for backoff

### PR DESCRIPTION
**Pull request by TimothyMothra**
Towards https://github.com/Azure/azure-sdk-for-net/issues/34581

### Changes
- log the interval whenever backoff occurs.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.cs [hunk 1]
```diff
@@ -268,7 +272,15 @@ public void TransmitterFailed(string instrumentationKey, Exception ex)
         public void TransmitterFailed(string instrumentationKey, string exceptionMessage) => WriteEvent(33, instrumentationKey, exceptionMessage);
 
         [NonEvent]
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        internal bool IsEnabled(EventLevel eventLevel) => IsEnabled(eventLevel, EventKeywords.All);
+        public void BackoffEnabled(TimeSpan backoffInterval)
```
Comment (reyang) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.cs:275]: Why is this needed?

**Review by rajkumar-rangaraj — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.cs (hunk 1 above)
Comment (TimothyMothra) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Diagnostics/AzureMonitorExporterEventSource.cs]: To help investigate "no data" scenarios.
The Exporter will backoff for a period of up to 1 hour.

**Review by vishweshbankwar — APPROVED** [review]
